acorn/src/tokenize.js

```diff
@@ -314,6 +314,20 @@ pp.readToken_question = function() { // '?'
   return this.finishOp(tt.question, 1)
 }
 
+pp.readToken_numberSign = function() { // '#'
+  const ecmaVersion = this.options.ecmaVersion
+  let code = "#"
```

Comment:
It seems `code` is treated like a number later on. Should the init value be the code point for `#` instead?

Reply:
Fixed in https://github.com/acornjs/acorn/pull/1026/commits/cc1a87ee2b2269238ce700b1b42a83adfd2e730e.